File: library/src/main/java/org/polaric/colorful/CActivity.java
package org.polaric.colorful;

import android.os.Build;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.WindowManager;

public abstract class CActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setTheme(Colorful.getThemeDelegate().getStyle());
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            if (Colorful.getThemeDelegate().isTranslucent()) {
                getWindow().addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
            }
        }
    }
}

File: library/src/main/java/io/multimoon/colorful/CActivity.kt
package io.multimoon.colorful

import android.annotation.TargetApi
import android.app.Activity
import android.os.Bundle
import android.support.annotation.RequiresApi

@RequiresApi(21)
open class CActivity : Activity(), CThemeInterface {

    override var themeString: String = ""

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        handleOnCreate(this, savedInstanceState, BaseTheme.THEME_MATERIAL)
    }

    override fun onResume() {
        super.onResume()
        handleOnResume(this)
    }
}